sklearn/naive_bayes.py

```diff
@@ -1425,3 +1436,574 @@ def _joint_log_likelihood(self, X):
+        which are calculated as relative frequencies, extracted from
+        subestimators, or provided, according to the value of `priors`
+        at initialization.
+
+    class_count_ : ndarray of shape (n_classes,)
+        Number of samples encountered for each class during fitting. This
+        value is weighted by the sample weight when provided.
+
+    n_classes_ : int
+        The number of classes known to the naive Bayes classifier, `n_classes`.
+
+    classes_ : ndarray of shape (n_classes,)
+        Class labels known to the classifier.
+
+    feature_names_in_ : ndarray of shape (`n_features_in_`,)
+        Names of features seen during :term:`fit`. Only defined if `X` has
+        feature names that are all strings.
+
+    See Also
+    --------
+    BernoulliNB : Naive Bayes classifier for multivariate Bernoulli models.
+    CategoricalNB : Naive Bayes classifier for categorical features.
+    ComplementNB : Complement Naive Bayes classifier.
+    MultinomialNB : Naive Bayes classifier for multinomial models.
+    GaussianNB : Gaussian Naive Bayes.
+    :class:`~sklearn.compose.ColumnTransformer` : Applies transformers to columns.
+
+    Notes
+    -----
+    ColumnwiseNB combines multiple naive Bayes estimators by expressing the
+    overall joint probability ``P(x,y)`` through ``P(x_i,y)``, the joint
+    probabilities of the subestimators::
+
+        ``Log P(x,y) = Log P(x_1,y) + ... + Log P(x_N,y) - (N - 1) Log P(y)``,
+
+    where ``N`` denotes ``n_estimators``, the number of estimators.
+    It is implicitly assumed that the class log priors are finite and agree
+    between the estimators and the subestimator::
+
+        ``- inf < Log P(y) = Log P(y|1) = ... = Log P(y|N)``.
+
+    The meta-estimators does not check if this condition holds. Meaningless
+    results, including ``NaN``, may be produced by ColumnwiseNB if the class
+    priors differ or contain a zero probability.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> rng = np.random.RandomState(1)
+    >>> X = rng.randint(5, size=(6, 100))
+    >>> y = np.array([0, 0, 1, 1, 2, 2])
+    >>> from sklearn.naive_bayes import MultinomialNB, GaussianNB, ColumnwiseNB
+    >>> clf = ColumnwiseNB(estimatorNBs=[('mnb1', MultinomialNB(), [0, 1]),
+    ...                                ('mnb2', MultinomialNB(), [3, 4]),
+    ...                                ('gnb1', GaussianNB(), [5])])
+    >>> clf.fit(X, y)
+    ColumnwiseNB(estimatorNBs=[('mnb1', MultinomialNB(), [0, 1]),
+                            ('mnb2', MultinomialNB(), [3, 4]),
+                            ('gnb1', GaussianNB(), [5])])
+    >>> print(clf.predict(X))
+    [0 0 1 0 2 2]
+    """
+
+    _required_parameters = ["estimatorNBs"]
+
+    def _log_message(self, name, idx, total):
+        if not self.verbose:
+            return None
+        return "(%d of %d) Processing %s" % (idx, total, name)
+
+    def __init__(self, estimatorNBs, *, priors=None, n_jobs=None, verbose=False):
```

Comment:
The capitalisation of `estimatorNBs` is quite unconventional. Can it be `estimators` or `handlers` or `base_nbs` or `nb_estimators`?

Reply:
Titanic dataset has been mention in #12957 in relation to mixing Gaussian and Bernoulli features in Naive Bayes. Do you think this will work as an example for the gallery?

Reply:
As for the capitalisation of `estimatorNBs`, I would personally prefer `estimators`, but I can't use this name at the moment. As I explained in **item 14** in the first message: some rely on the presence of `estimators` in parameters as a criterion for duck-typing. Do you think I should create an issue to remove this duck-typing from the tests?